core/store/migrations/migration1565291711/migrate.go

```diff
@@ -0,0 +1,169 @@
+	`).Error; err != nil {
+			return errors.Wrap(err, "failed to update job_spec_id id on task_specs")
+		}
+
+		if err := tx.Exec(`
+ALTER TABLE service_agreements DROP COLUMN "job_spec_id";
+ALTER TABLE service_agreements RENAME COLUMN "job_spec_id_uuid" TO "job_spec_id";
+
+ALTER TABLE service_agreements ADD CONSTRAINT "service_agreements_job_spec_id_fkey" FOREIGN KEY ("job_spec_id") REFERENCES job_specs ("id") ON DELETE CASCADE;
+	`).Error; err != nil {
+			return errors.Wrap(err, "failed to update job_spec_id id on service_agreements")
+		}
+
+		if err := tx.Exec(`
+ALTER TABLE job_runs ADD COLUMN "id_uuid" uuid;
+UPDATE job_runs
+SET
+	"id_uuid" = CAST("id" as uuid);
+ALTER TABLE job_runs DROP CONSTRAINT "job_runs_pkey" CASCADE;
+	`).Error; err != nil {
+			return errors.Wrap(err, "failed to add id_uuid on job_runs")
+		}
+
+		if err := tx.Exec(`
+ALTER TABLE task_runs ADD COLUMN "job_run_id_uuid" uuid;
+UPDATE task_runs
+SET
+	"job_run_id_uuid" = CAST("job_run_id" as uuid);
+	`).Error; err != nil {
+			return errors.Wrap(err, "failed to add job_run_id_uuid on task_runs")
+		}
+
+		if err := tx.Exec(`
+ALTER TABLE link_earned ADD COLUMN "job_run_id_uuid" uuid;
+UPDATE link_earned
+SET
+	"job_run_id_uuid" = CAST("job_run_id" as uuid);
+	`).Error; err != nil {
+			return errors.Wrap(err, "failed to add job_run_id_uuid on link_earned")
+		}
+
+		if err := tx.Exec(`
+ALTER TABLE job_runs DROP COLUMN "id";
+ALTER TABLE job_runs RENAME COLUMN "id_uuid" TO "id";
+ALTER TABLE job_runs ADD CONSTRAINT "job_run_pkey" PRIMARY KEY ("id");
+	`).Error; err != nil {
+			return errors.Wrap(err, "failed to rename id on job_runs")
+		}
+
+		if err := tx.Exec(`
+ALTER TABLE task_runs DROP COLUMN "job_run_id";
+ALTER TABLE task_runs RENAME COLUMN "job_run_id_uuid" TO "job_run_id";
+
+ALTER TABLE task_runs ADD CONSTRAINT "task_runs_job_run_id_fkey" FOREIGN KEY ("job_run_id") REFERENCES job_runs ("id") ON DELETE CASCADE;
+	`).Error; err != nil {
+			return errors.Wrap(err, "failed to update job_run_id id on task_runs")
+		}
+
+		if err := tx.Exec(`
+ALTER TABLE link_earned DROP COLUMN "job_run_id";
+ALTER TABLE link_earned RENAME COLUMN "job_run_id_uuid" TO "job_run_id";
+
+ALTER TABLE link_earned ADD CONSTRAINT "link_earned_job_run_id_fkey" FOREIGN KEY ("job_run_id") REFERENCES job_runs ("id") ON DELETE CASCADE;
+	`).Error; err != nil {
+			return errors.Wrap(err, "failed to update job_run_id id on link_earned")
+		}
+
+		if err := tx.Exec(`
+ALTER TABLE task_runs ADD COLUMN "id_uuid" uuid;
+UPDATE task_runs SET "id_uuid" = CAST("id" as uuid);
+ALTER TABLE task_runs DROP COLUMN "id";
+ALTER TABLE task_runs RENAME COLUMN "id_uuid" TO "id";
+	`).Error; err != nil {
+			return errors.Wrap(err, "failed to update id on task_runs")
+		}
+	} else {
+		if err := tx.Exec(`
+CREATE TABLE IF NOT EXISTS "link_earned_new" (
+	"id" integer primary key autoincrement NOT NULL,
+	"job_run_id" varchar(36) REFERENCES job_runs(id) NOT NULL UNIQUE,
```

Comment:
Instead of `varchar` should this be `uuid`?

```
"job_run_id" uuid REFERENCES job_runs(id) NOT NULL UNIQUE,
```

Reply:
Sqlite does not support uuid.

Reply:
This is behind `if dbutil.IsPostgres(tx) {` so it isn't run on Sqlite. This whole migration is essentially postgres only so I would take use `uuid` liberally.

Reply:
Ah, apologies. This is in the else so sqlite. 👍